core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -81,6 +81,15 @@ class DAGSchedulerEventProcessLoopTester(dagScheduler: DAGScheduler)
     }
   }
 
+  def postInCurrentThread(event: DAGSchedulerEvent): Unit = {
```

Comment:
This is not how the DAGScheduler processes the events. 
The production code does it in a dedicated single thread for event processing called `dag-scheduler-event-loop`. 
See `DAGSchedulerEventProcessLoop` for details.

This way this reproduction is invalid.

Reply:
@attilapiros .. It is right that DagScheduler processes the events sequentially, but in case of failure, it delegates it to another thread pool, causing asynchronicity.
The reason for the test using asynchroncity in the even loop, is just to be able to simulate the issue with the given constraints of being unable to control the behaviour of the  event pool emebedded withthin the DagScheduler itself.